Automattic/simplenote-macos
# Email Verification UI

### Details
In this PR we're building macOS counterpart of [Simplenote iOS PR 1081](https://github.com/Automattic/simplenote-ios/pull/1081).

@eshurakov Apologies it goes beyond the 500 LOC limit, the changes required to stylize NSButton alone add up to ~90-ish LOC 🤦 

Thanks in advance!! 🥳 

Ref. #800

### Test: Basic UI
1. Log into your account and exit Simplenote
2. Checkout PR 5e5e3e3 (Contains testing code!)
3. Relaunch the App

- [x] Verify that the new Verification UI shows up after 2 seconds
- [x] Verify that the new UI looks great when you switch from/to Dark/Light mode
- [x] Verify that pressing **ESC** dismisses the new Alert

### Test: Email Update
Repeat steps 1-3 from the previous test

- [x] Verify that pressing **Change Account Email** opens the Settings UI on the web
- [x] Verify the new UI gets dismissed

### Test: Confirm Flow
Repeat steps 1-3 from the first test

- [x] Verify that pressing **Confirm** causes a Spinner to show up, and locks the screen
- [x] Verify that after a few seconds the UI switches to **Email Verification** mode
- [x] Verify that pressing Resend causes the spinner to show up again

### Release
`RELEASE-NOTES.txt` was updated in e223b42 with:

> New account verification screen


### Screenshots
<img width="300" alt="Dark" src="https://user-images.githubusercontent.com/1195260/105412670-cbe74f00-5c13-11eb-8d5d-d644df5cba28.png"> <img width="300" alt="Light" src="https://user-images.githubusercontent.com/1195260/105412646-c7229b00-5c13-11eb-9e47-546001f21d38.png">
